core/loadpoint.go

```diff
@@ -703,9 +703,9 @@ func (lp *Loadpoint) syncCharger() error {
 
 	// #2: sync charger
 	switch {
-	case enabled == lp.enabled:
+	case enabled && lp.enabled:
```

Comment:
Nice catch! 
Took me a moment to verify this is OK, but you are right.

Reply:
Yeah, me too. Explicitly mentioned it after verifying 3 times since it's not obvious.

Reply:
I like this PR - funny enough I just stumbled over the exact same issue but had no good idea on how to fix it properly.
Anyways: The change misses the case where both lp and charger are not enabled but phases differ. Of course this can only happen for chargers that implement PhaseGetter and in rare cases (i. E. when the charger reboots). But it might be better to keep the old logic here (sync phases when lp and charger agree on the enabled state, sync current only when both are enabled).

Reply:
> But it might be better to keep the old logic here (sync phases when lp and charger agree on the enabled state, sync current only when both are enabled).

Point of the discussion above it that this *is* the old logic since the `&& enabled` condition was part of every single if statement below if you look carefully?

That does of course not mean that there isn't another condition that the old version did not address yet (and still doesn't after this PR).

Reply:
Good point - I missed the one enabled in the "sync phases" condition. Indeed for this one indeed it **is** the old behavior. 
Looking closer the change has an issue: Now, if charger and loadpoint are both disabled, the other case statements are evaluated - which is wrong. It has to be `case enabled == lp.enabled:` to avoid that

Reply:
Autsch. @MarkusGH das wäre dann

```
        // #2: sync charger
        switch {
-       case enabled && lp.enabled:
+       case enabled == lp.enabled:
+               if !enabled {
+                       return nil
+               }
+
```

Reply:
Würde den Zweck erfüllen - wobei ich es schöner fände, die Bedingung einfach wieder hier einzufügen:

`if enabled && charger, ok := lp.charger.(api.CurrentGetter); ok {`

Die Phasen kann man ja auch bei einem nicht enabelten Charger synchronisieren.

Reply:
Ist leider nicht so einfach wg. des `else`. Siehe https://github.com/evcc-io/evcc/pull/14639. Am disabled state würde ich erstmal nix ändern- der PR sollte ausser den Hinzufügungen erstmal nichts am Verhalten ändern.